crap-advisor-frontend/src/core/components/RestroomMarker/index.jsx

```diff
@@ -4,15 +4,22 @@ import L from "leaflet";
 import restroomImage from "../../../assets/restroomMarker.png";
 import "./index.css"
 
-const RestroomMarker = ({restroom}) => {
+const RestroomMarker = ({restroom, selectRestroom}) => {
     const icon = L.divIcon({
         className: "icon",
-        html: `<div class="marker-container">` +
-            `<img class="marker-icon" src='${restroomImage}' alt='Toilet'/><p class="marker-text outlined">${restroom.rating}</p></div>`
+        html:
+            `<div class="marker-container">` +
+            `<img class="marker-icon" src='${restroomImage}' alt='Toilet'/>` +
+            `<p class="marker-text outlined">${restroom.rating != null ? restroom.rating : ""}</p>` +
+            `</div>`
     })
 
     return (
-        <Marker position={[restroom.position.latitude, restroom.position.longitude]} icon={icon}>
+        <Marker
+            position={[restroom.location.latitude, restroom.location.longitude]}
+            icon={icon}
+            eventHandlers={{click: () => selectRestroom(restroom)}}
```

Comment:
Move the function into a variable with an anonymous void function, because it's will be rendered in virtualDOM twicely.

Reply:
Didn't understand a single word about this virtual thing, but moved the function anyway